x-pack/plugins/infra/common/dependency_mocks/index_patterns.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from, of } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import {
+  fieldList,
+  FieldSpec,
+  IIndexPattern,
+  IndexPattern,
+  IndexPatternsContract,
+  RuntimeField,
+} from 'src/plugins/data/common';
+
+type IndexPatternMock = Pick<
+  IndexPattern,
+  | 'fields'
+  | 'getComputedFields'
+  | 'getFieldByName'
+  | 'getTimeField'
+  | 'id'
+  | 'isTimeBased'
+  | 'title'
+  | 'type'
+>;
+type IndexPatternMockSpec = Pick<IIndexPattern, 'id' | 'title' | 'type' | 'timeFieldName'> & {
+  fields: FieldSpec[];
+};
+
+export const createIndexPatternMock = ({
```

Comment:
👍  For moving this to `common`. I'll try to bear in mind interoperability between Jest and Storybooks.